addons/binding/org.openhab.binding.mysensors/src/main/java/org/openhab/binding/mysensors/handler/MySensorsThingHandler.java

```diff
@@ -0,0 +1,396 @@
+ *
+ * @author Tim Oberföll
+ */
+public class MySensorsThingHandler extends BaseThingHandler implements MySensorsGatewayEventListener {
+
+    private Logger logger = LoggerFactory.getLogger(getClass());
+
+    private MySensorsSensorConfiguration configuration = null;
+
+    private int nodeId = 0;
+    private int childId = 0;
+    private boolean requestAck = false;
+    private boolean revertState = true;
+
+    private boolean smartSleep = false;
+
+    private int expectUpdateTimeout = -1;
+
+    private DateTimeType lastUpdate = null;
+
+    private MySensorsGateway myGateway;
+
+    public MySensorsThingHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        configuration = getConfigAs(MySensorsSensorConfiguration.class);
+        nodeId = Integer.parseInt(configuration.nodeId);
+        childId = Integer.parseInt(configuration.childId);
+        requestAck = configuration.requestAck;
+        revertState = configuration.revertState;
+        smartSleep = configuration.smartSleep;
+        expectUpdateTimeout = configuration.childUpdateTimeout;
+
+        logger.debug(configuration.toString());
+
+        myGateway = getBridgeHandler().getMySensorsGateway();
+        addIntoGateway(getThing(), configuration);
+
+        registerListeners();
+
+        updateStatus(ThingStatus.ONLINE);
+    }
+
+    @Override
+    public void dispose() {
+        myGateway.removeEventListener(this);
+        super.dispose();
+    }
+
+    @Override
+    public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
+        logger.debug("MySensors Bridge Status updated to {} for device: {}", bridgeStatusInfo.getStatus().toString(),
+                getThing().getUID().toString());
+        if (bridgeStatusInfo.getStatus().equals(ThingStatus.ONLINE)
+                || bridgeStatusInfo.getStatus().equals(ThingStatus.OFFLINE)) {
+
+            if (bridgeStatusInfo.getStatus().equals(ThingStatus.ONLINE)) {
+                registerListeners();
+            } else {
+                myGateway.removeEventListener(this);
+            }
+
+            // the node has the same status of the bridge
+            updateStatus(bridgeStatusInfo.getStatus());
+        }
+    }
+
+    @Override
+    public void handleConfigurationUpdate(Map<String, Object> configurationParameters) {
+        logger.debug("Configuation update fo thing {}-{}: {}", nodeId, childId, configurationParameters);
+        super.handleConfigurationUpdate(configurationParameters);
+    }
+
+    @Override
+    public void handleUpdate(ChannelUID channelUID, State newState) {
+        logger.trace("New state {} received for channel uid {}", newState, channelUID);
+        super.handleUpdate(channelUID, newState);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.trace("Command {} received for channel uid {}", command, channelUID);
+        /*

```

Comment:
I think that speaks from the code but if you want to say it you could better use 
```java
// We don't handle refresh commands yet
```